src/metatensor/models/experimental/soap_bpnn/model.py

```diff
@@ -291,27 +281,46 @@ def forward(
 
         soap_features = self.layernorm(soap_features)
 
-        hidden_features = self.bpnn(soap_features)
+        last_layer_features = self.bpnn(soap_features)
+
+        # output the hidden features, if requested:
+        if "last_layer_features" in outputs.keys():
+            last_layer_features_options = outputs["last_layer_features"]
+            out_features = last_layer_features.keys_to_samples("center_type")
+            if last_layer_features_options.per_atom:
+                # this operation should just remove the center_type label
+                return_dict["last_layer_features"] = metatensor.torch.sum_over_samples(
+                    out_features, ["center_type"]
+                )
+            else:
+                return_dict["last_layer_features"] = metatensor.torch.sum_over_samples(
```

Comment:
Should we offer an option to return mean_over_samples?

We should at least specify somewhere that the model in itself will only support "raw" or "sum" of last layer features per structures, and so if you want the mean over samples, you should be computing that yourself.

Reply:
I would tend for the second solution here. The place to document these things is https://lab-cosmo.github.io/metatensor/latest/atomistic/outputs.html!